feat(faceted-search) configurable text badge placeholder

Nigui opened the pull request:
**What is the problem this PR is trying to solve?**

We want to improve user experience of faceted-search users. Currently faceted search text badge value can be edited through an input in a popover form. 

This PR makes input placeholder configurable from component consumer. 

**What is the chosen solution to this problem?**

Make faceted search text badge input placeholder configurable. 

`BadgeTextForm` component expose a new property called `placeholder` which is initialized by `BadgeText` component through its new property `inputTextPlaceholder`

**Please check if the PR fulfills these requirements**

- [X] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [X] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

<!-- You can add more checkboxes here -->

**[ ] This PR introduces a breaking change**

<!-- if the PR introduces a breaking change, add the description here. So when you merge this PR, add this description into the [breaking change wiki](https://github.com/Talend/ui/wiki/BREAKING-CHANGE) in the next version -->

<!-- **Original Template** -->

<!-- https://github.com/Talend/tools/blob/master/tools-root-github/.github/PULL_REQUEST_TEMPLATE.md -->

Nigui commented on packages/faceted-search/src/components/Badges/BadgeText/BadgeText.component.js:
```diff
@@ -38,6 +38,7 @@ const BadgeText = ({
 	readOnly,
 	removable,
 	displayType,
+	inputTextPlaceholder,
```
Was just precautions if other inputs are added to component. But I can simplify 👍

jsomsanith-tlnd commented on packages/faceted-search/stories/badgesDefinitions.js:
```diff
@@ -33,10 +34,10 @@ export const badgeConnectionName = {
 		badgePerFacet: 'N',
 		entitiesPerBadge: '1',
 		operators: ['containsIgnoreCase', 'equals', 'notEquals', 'match a regexp'],
+		inputTextPlaceholder: 'Enter a connection name',
```
indeed, the input attributes seem to be located in the properties part, and yes it is not that clear what the parts contain 🤔 
+1 to move it though

Nigui commented:
I've updated multiple things : 
- renamed text badge placeholder property from `inputTextPlaceholder` to `placeholder`
- moved badge type properties from `metadata` to `properties` in story
- edited a utility function which builds badge definition from a raw object, adding a new property in raw object called `typeProperties` aiming to pass badge type properties in the right way. 
- added tests to this utility function
